src/main/java/com/grafana/loki/LokiSourceTask.java

```diff
@@ -0,0 +1,103 @@
+package com.grafana.loki;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.apache.hc.client5.http.fluent.Request;
+import org.apache.hc.core5.net.URIBuilder;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.source.SourceTask;
+import org.apache.kafka.connect.source.SourceTaskContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LokiSourceTask extends SourceTask {
+
+  private static final Logger log = LoggerFactory.getLogger(LokiSourceTask.class);
+
+  public static final String TIMESTAMP_FIELD = "timestamp";
+
+  private String lokiEndpoint;
+  private String lokiQuery;
+  private String topic;
+  private Long lastTimestamp;
+
+  @Override
+  public String version() {
+    return new LokiSourceConnector().version();
+  }
+
+  @Override
+  public void start(Map<String, String> props) {
+    lokiEndpoint = props.get(LokiSourceConnector.ENDPOINT_CONFIG);
+    lokiQuery = props.get(LokiSourceConnector.QUERY_CONFIG);
+    topic = props.get(LokiSourceConnector.TOPIC_CONFIG);
+  }
+
+  @Override
+  public void initialize(SourceTaskContext context) {
+    super.initialize(context);
+    lastTimestamp = 0L;
+    context.offsetStorageReader().offset(Collections.emptyMap());
+  }
+
+  @Override
+  public synchronized void stop() {}
+
+  @Override
+  public List<SourceRecord> poll() throws InterruptedException {
```

Comment:
we do not throw `InterruptedException`. so we can remove it from the declaration here.